src/components/Calendar.vue

```diff
@@ -86,32 +89,26 @@ export default {
   computed: {
     sortedTimes () {
       return this.sortTimes()
+    },
+    hasUserSchedule () {
+      return !_.isEmpty(this.availability)
     }
   },
-  beforeMount () {
-    var originalAvailability = {}
+  async beforeMount () {
```

Comment:
Moving these requests into an earlier lifecycle method (`created`) and awaiting `all` would be preferable (lower latency).
```diff
+  created () {
     if (!this.user.hasSchedule) {
       CalendarService.initAvailability(this, this.user._id)
     }
-    var originalAvailability = await CalendarService.getAvailability(this, this.user._id)
-    var userTimezone = await CalendarService.getTimezone(this, this.user._id)
-    if (userTimezone && this.userTzInList(userTimezone)) {
-      this.selectedTz = userTimezone
-    } else {
-      this.selectedTz = moment.tz.guess()
-    }
-    var estNow = moment.tz('America/New_York').hour()
-    var userNow = moment.tz(this.selectedTz).hour()
-    var offset = userNow - estNow
-    this.availability = this.convertAvailability(originalAvailability, offset)
+
+    var originalAvailability = CalendarService.getAvailability(this, this.user._id)
+    var userTimezone = CalendarService.getTimezone(this, this.user._id)
+
+    Promise
+      .all([originalAvailability, userTimezone])
+      .then(([originalAvailability, userTimezone]) => {
+        if (userTimezone && this.userTzInList(userTimezone)) {
+          this.selectedTz = userTimezone
+        } else {
+          this.selectedTz = moment.tz.guess()
+        }
+        var estNow = moment.tz('America/New_York').hour()
+        var userNow = moment.tz(this.selectedTz).hour()
+        var offset = userNow - estNow
+        this.availability = this.convertAvailability(originalAvailability, offset)
+      })

```

Reply:
Ah yeah 👍  I'm seeing this as the suggested pattern for fetching after navigation: https://router.vuejs.org/guide/advanced/data-fetching.html

The other suggested pattern in that guide is for fetching data before route navigation in the `beforeRouteEnter` hook. Could see this working too, since we don't really have anything to show without the user's schedule data. But it sounds like this requires implementing our own "loading view" because Vue just stays on the previous route and does nothing until the data comes back.. So I'll go with the `created` method for now :)